src/Response.php

```diff
@@ -186,6 +180,19 @@ private function setStatusCode($code)
                 static::MAX_STATUS_CODE_VALUE
             ));
         }
+
+        if (! is_string($reasonPhrase)) {
+            throw new InvalidArgumentException(sprintf(
+                'Unsupported HTTP method; must be a string, received %s',
```

Comment:
"Unsupported HTTP method"?

I'm thinking you mean "Unsupported response reason phrase" here.